### PR TITLE
EVG-1964 db and models for degraded mode

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

**Comment by john-m-liu** [comment]
This is the second patch set for EVG-1964, but this one has the branch create correctly unlike before

File: model/admin/db.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+var settingsQuery = db.Query(bson.M{IdKey: systemSettingsID})
+
+func GetSettingsFromDB() (*AdminSettings, error) {
+	settings := &AdminSettings{}
+	query := db.Q{}
+	query = query.Filter(settingsQuery)
+	err := db.FindOneQ(Collection, query, settings)
+	if err != nil {
+		// if the settings document doesn't exist, return the defaults
```
Comment (john-m-liu) [inline model/admin/db.go]: I considered this, but it felt cleaner to me to return the defaults. The document will get inserted the first time someone changes one of the settings

**Comment by john-m-liu** [comment]
I address the comments in the most recent commit

File: model/admin/db.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+// GetSettingsFromDB retrieves the admin settings document. If no document is
+// present in the DB, it will return the defaults
+func GetSettingsFromDB() (*AdminSettings, error) {
+	settings := &AdminSettings{}
+	query := db.Q{}
+	query = query.Filter(settingsQuery)
+	err := db.FindOneQ(Collection, query, settings)
+	if err != nil {
+		// if the settings document doesn't exist, return the defaults
+		if err.Error() == "not found" {
```
Comment (tychoish) [inline model/admin/db.go]: this seems fragile.

File: model/admin/db.go (hunk 1 above)
Comment (tychoish) [inline model/admin/db.go]: this seems fine.

File: model/admin/db.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/db/bsonutil"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	Collection          = "admin"
+	systemSettingsDocID = "0"
```
Comment (tychoish) [inline model/admin/db.go]: do we want a more explicit name for this?

Comment (john-m-liu) [inline model/admin/db.go]: Changed to "global"

**Pull request merged**